### PR TITLE
Clean up tests

### PR DESCRIPTION
**Pull request by CodeSandwich**
Refactor to use abstract classes to organize tests by splitting `EthPool.t.sol` into `EthPool.t.sol`, `Pool.t.sol` and `UserUtils.t.sol`.

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@ let
   pkgs = import (builtins.fetchGit rec {
     name = "dapptools-${rev}";
     url = https://github.com/dapphub/dapptools;
-    rev = "afbb707102baa77eac6ad70873fcd3c59a2ff53c";
+    rev = "e41b6cd9119bbd494aba1236838b859f2136696b";
```
Comment (manuelpolzhofer) [inline shell.nix:5]: nice, 12 days ago. :)

File: src/test/Pool.t.sol [hunk 2]
```diff
@@ -0,0 +1,379 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.7;
+
+import {DSTest} from "ds-test/test.sol";
+import {PoolUserUtils} from "./UserUtils.t.sol";
+import {PoolUser} from "./User.t.sol";
+import {Hevm} from "./BaseTest.t.sol";
+import {Pool, ReceiverWeight} from "../Pool.sol";
+
+abstract contract PoolTest is PoolUserUtils {
```
Comment (manuelpolzhofer) [inline src/test/Pool.t.sol:10]: Ah nice, so the tests won't be executed on the abstract contract

**Review by manuelpolzhofer — APPROVED** [review]
Nice.

I can't see the CI run for this PR.

**Comment by CodeSandwich** [comment]
@manuelpolzhofer GitHub actions apparently get confused when rebasing a branch and don't trigger `on.pull_request`. I've pushed some dummy commits to trigger it, I'll remove them when merging.

**Pull request merged**